### PR TITLE
[+] create architecture for ClientCommands part

### PR DESCRIPTION
**Pull request by osvegn**
Pull request link to the issue #62.

### CODE REVIEW

**Review by Cavonstavant — APPROVED** [review]
LGTM!

**Review by LucasTesnier — APPROVED** [review]
NTR!

File: server/src/CMakeLists.txt [hunk 1]
```diff
@@ -14,6 +14,9 @@ add_executable(${TARGET} ${SRC})
 #     PRIVATE ${SRC_ROOT}
 # )
 
+target_link_libraries(${TARGET}
+    PUBLIC ${PROJECT_NAME}_commands
```
Comment (Cavonstavant) [inline server/src/CMakeLists.txt]: why public

**Review by Cavonstavant — APPROVED** [review]

File: server/src/CMakeLists.txt (hunk 1 above)
Comment (osvegn) [reply server/src/CMakeLists.txt]: because

**Pull request merged**